api/src/modules/Session/model.ts

```diff
@@ -0,0 +1,142 @@
+    t.nonNull.date('createdAt');
+    t.nonNull.string('createdBy');
+    t.date('referenceExpiryDate');
+    t.nonNull.string('referenceTokenId');
+    t.nonNull.boolean('isUserCreated');
+    t.string('name');
+    t.date('revokedAt');
+  },
+});
+
+export const SessionCreate = inputObjectType({
+  name: 'SessionCreate',
+  definition(t) {
+    t.int('expiryDurationSeconds');
+    t.nonNull.string('name');
+  },
+});
+
+export const SessionCreateOutput = objectType({
+  name: 'SessionCreateOutput',
+  definition(t) {
+    t.nonNull.field('session', { type: 'Session' });
+    t.nonNull.string('token');
+  },
+});
+
+async function newSession(
+  prisma: PrismaClient,
+  session: Prisma.SessionCreateInput,
+) {
+  return prisma.session.create({
+    data: session,
+  });
+}
+
+const generateTokenAndSession = async (
+  prisma: PrismaClient,
+  userId: string,
+  session: { expiryDurationSeconds?: number | null; name: string },
+  isUserCreated: boolean = false,
+) => {
+  const createId = randomUUID();
+  const createdToken = tokenUtils.generate(createId, session.expiryDurationSeconds);
+  const formattedToken = tokenUtils.format(createdToken);
+  const createData = {
+    name: session.name,
+    referenceExpiryDate: tokenUtils.getExpiryDate(session.expiryDurationSeconds),
+    id: createId,
+    referenceTokenId: formattedToken,
+    isUserCreated,
+    creator: {
+      connect: {
+        id: userId,
+      },
+    },
+  };
+  const createdSession = await newSession(prisma, createData);
+  return {
+    token: createdToken,
+    session: createdSession,
+  };
+};
+
+export function getSessionCrud(prisma: PrismaClient) {
+  return {
+    listSessions: async (userId: string) => prisma.session.findMany({
+      where: {
+        createdBy: userId,
+      },
+    }),
+    revoke: async (sessionId: string, userId: string) => {
+      try {
+        return await prisma.session.update({
+          where: {
+            createdBy_id: {
+              id: sessionId,
+              createdBy: userId,
+            },
+          },
+          data: {
+            revokedAt: new Date(),
+          },
+        });
+      } catch (e) {
+        throw new GraphQLError('Failed to update session', { extensions: { code: 'SESSION_UPDATE_FAILED' } });
+      }
+    },
+    createSignInSession: async (userId: string) => generateTokenAndSession(
+      prisma,
+      userId,
+      { expiryDurationSeconds: ms(JWT_EXPIRATION_PERIOD) / 1000, name: 'Sign in' },
+    ),
+    createSignUpSession: async (userId: string) => generateTokenAndSession(
+      prisma,
+      userId,
+      { expiryDurationSeconds: ms(JWT_EXPIRATION_PERIOD) / 1000, name: 'Sign up' },
+    ),
+    createCustomSession: async (
+      userId: string,
+      session: { expiryDurationSeconds?: number | null; name: string },
+      isUserCreated: boolean = false,
+    ) => generateTokenAndSession(prisma, userId, session, isUserCreated),
+    async getUser(
+      token?: string,
+    ) {
```

Comment:
I suggested two options:
- `user.getUserByToken(token)`
- `session.getSessionByToken(token)`

And you made `session.getUser()` instead without bringing any arguments. Can you please elaborate why did you need to mix session with user?
```suggestion
    async getSessionByToken(
      token?: string,
    ) {
```